### DISCUSSION TITLE
Idea to reduce invalid PRs to this repo

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [daattali/beautiful-jekyll#1400] [Ideas]
I have an idea to reduce invalid PRs to this repo.

Create a template repo, that has a predefined config file with the `remote_theme` option already set https://beautifuljekyll.com/getstarted/#install-steps-hard

Then users could easily use the template (with no fork link back to the template or to this repo). Here is the official documentation on template repos: https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository

Here is a really simple example template repo: https://github.com/LizardByte/template-base

![image](https://github.com/user-attachments/assets/bb20417e-8540-437d-a6a6-9f8c84effcd5)

**Comment by daattali** [comment]
Thanks for the suggestion. I've thought about this in the past, but ultimately what keeps me from doing that is the fact that with forks, you're able to easily "sync" with the original repo (pull new changes). There's no way to do that with template repos, is there?

**Reply by ReenigneArcher** [reply]
I don't think you can do that with a template, but I also believe it won't work in a fork once you have modified things to not match the upstream (since GitHub can't resolve the conflicts).

**Reply by daattali** [reply]
I'm not sure what would happen with merge conflicts, whether GitHub can handle that with a nice UI or not. However, the vast majority of simple users that I've observed don't do anything that would cause a conflict. Most users add pages to `_posts` and edit the config file. A conflict is most likely to occur in the config file, but even there it's uncommon. A very slim fraction of people edit any other files, and usually these type of people are capable enough to know how to handle conflicts. Given all that, I do place a much higher importance on ease of updating, and since the sync feature works pretty well, I'd rather stay with it.
